src/tests/JNIGridnetClientSelfPlay.java

```diff
@@ -155,24 +189,30 @@ public void gameStep(int[][] action1, int[][] action2) throws Exception {
         }
     }
 
+    /**
+     * @param player
+     * @return Legal actions mask for given player.
+     * @throws Exception
+     */
     public int[][][] getMasks(int player) throws Exception {
         for (int i = 0; i < masks[0].length; i++) {
             for (int j = 0; j < masks[0][0].length; j++) {
-                for (int k = 0; k < masks[0][0][0].length; k++) {
-                    masks[player][i][j][k] = 0;
-                }
+            	Arrays.fill(masks[player][i][j], 0);
```

Comment:
may make it easier for the JIT compiler to recognise the code pattern and optimise to use `memset` or SIMD instructions.